### PR TITLE
Fixes #11409

### LINKED ISSUE #11409: If someone is gibbed while in a DNA scanner, the DNA scanner doesn't properly update
This means it can be ejected to drag them back from qdeletion, which gets you a dense torso with `invisibility` 101, meaning that other than pushing it around, it can't even be interacted with without admin intervention.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #11409 and anything else with an occupant variable that didn't handle this

### CODE REVIEW

File: code/game/dna/dna_modifier.dm [hunk 1]
```diff
@@ -114,7 +114,8 @@
 
 	if(!occupant)
 		for(var/mob/M in src)//Failsafe so you can get mobs out
-			M.forceMove(get_turf(src))
+			if(!(M.gcDestroyed))
```
Comment (PJB3005) [inline code/game/dna/dna_modifier.dm]: Redundant parentheses.

File: code/game/dna/dna_modifier.dm [hunk 2]
```diff
@@ -274,8 +274,9 @@
 	if(locked)
 		visible_message("Can't eject occupants while \the [src] is locked.")
 		return 0
-	occupant.forceMove(exit)
-	occupant.reset_view()
+	if(!(occupant.gcDestroyed))
```
Comment (PJB3005) [inline code/game/dna/dna_modifier.dm]: Again.

File: code/game/machinery/Sleeper.dm [hunk 3]
```diff
@@ -530,8 +530,9 @@
 		if(x in component_parts)
 			continue
 		x.forceMove(src.loc)
-	occupant.forceMove(exit)
-	occupant.reset_view()
+	if(!(occupant.gcDestroyed))
```
Comment (PJB3005) [inline code/game/machinery/Sleeper.dm]: 🤔

File: code/game/machinery/adv_med.dm [hunk 4]
```diff
@@ -219,12 +219,12 @@
 			continue
 		x.forceMove(src.loc)
 
-	src.occupant.forceMove(exit)
-	src.occupant.reset_view()
-	src.occupant = null
+	if(!(occupant.gcDestroyed))
```
Comment (PJB3005) [inline code/game/machinery/adv_med.dm]: Again.

File: code/game/machinery/rechargestation.dm [hunk 5]
```diff
@@ -238,11 +238,12 @@
 		return
 	//for(var/obj/O in src)
 	//	O.loc = src.loc
-	if (src.occupant.client)
-		src.occupant.client.eye = src.occupant.client.mob
-		src.occupant.client.perspective = MOB_PERSPECTIVE
-	src.occupant.forceMove(src.loc)
-	src.occupant = null
+	if(!(occupant.gcDestroyed))
```
Comment (PJB3005) [inline code/game/machinery/rechargestation.dm]: Again.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/dna/dna_modifier.dm (hunk 1 above)
Comment (gbasood) [reply code/game/dna/dna_modifier.dm]: a world without parens is scary
the irish put everything they write in parens to avoid triggering eachother

Comment (Exxion) [reply code/game/dna/dna_modifier.dm]: Everyone knows more parentheses means more safety

<sub>--An actual comment in my old Steambot's code</sub>

**Comment by gbasood** [comment]
@DrCelt reeee
fix the parens or check the box that lets contributors make changes to your PR

**Comment by Intigracy** [comment]
It starts ticked by default, he intentionally unticked it.

**Comment by 9600bauds** [comment]
is this really what passes as a negative review 22 days ago smh

**Comment by ghost** [comment]
. NM

**Comment by Probe1** [comment]
But all you had to do is check let collabs make edits

**Comment by ghost** [comment]
Okay so either I closed this due to a multiple personality or I assdialed the closing of the PR

**Comment by ghost** [comment]
It's ticked

**Comment by gbasood** [comment]
weird I'm not seing the option to pull it
at least Probe does though

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**